rosidl_typesupport_introspection_c/resource/msg__type_support.c.em

```diff
@@ -154,7 +154,7 @@ for index, field in enumerate(spec.fields):
     # bool is_array_
     print('    %s,  // is array' % ('true' if field.type.is_array else 'false'))
     # size_t array_size_
-    print('    %u,  // array size' % (field.type.array_size if field.type.array_size and not field.type.is_upper_bound else 0))
```

Comment:
For bounded dynamic arrays the introspection information must report the `array_size` as not `0` but as the *upper bound*.